src/SignalRServiceExtension/Bindings/SignalRAsyncCollector.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    public class SignalRAsyncCollector<T> : IAsyncCollector<T>
+    {
+        private readonly IAzureSignalRSender client;
+        private readonly string hubName;
+        private readonly SignalROutputConverter converter;
+
+        internal SignalRAsyncCollector(IAzureSignalRSender client, string hubName)
+        {
+            this.client = client;
+            this.hubName = hubName;
+            converter = new SignalROutputConverter();
+        }
+
+        public async Task AddAsync(T item, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var convertItem = converter.ConvertToSignalROutput(item);
+
+            if (convertItem.GetType() == typeof(SignalRMessage))
```

Comment:
Will convertedItem be null?

Reply:
Depends on how user pass the object when call us. Let me add null check and throw exceptions.

Reply:
And convertedItem won't be null because inner exception will throw before goes here.